android-kotlin/app/src/main/java/com/mongodb/app/ItemAdapter.kt

```diff
@@ -17,17 +16,52 @@ import org.bson.types.ObjectId
  * for a RecyclerView to display Realm objects on screen to a user.
  */
 internal class ItemAdapter(data: OrderedRealmCollection<Item>, val user: io.realm.mongodb.User, private val partition: String) : RealmRecyclerViewAdapter<Item, ItemAdapter.ItemViewHolder?>(data, true) {
+
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
         val itemView: View = LayoutInflater.from(parent.context).inflate(R.layout.item_view, parent, false)
         return ItemViewHolder(itemView)
     }
 
     override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
         val obj: Item? = getItem(position)
+        holder.id = obj?._id
         holder.name.text = obj?.name
+
+        holder.itemView.setOnClickListener {
+            val popup = PopupMenu(holder.itemView.context, holder.menu)
+            val menu = popup.menu
+            val deleteCode = -1
+            menu.add(0, deleteCode, Menu.NONE, "Delete")
+            popup.setOnMenuItemClickListener { menuItem: MenuItem? ->
+                when (menuItem!!.itemId) {
+                    deleteCode -> {
+                        deleteItem(holder.id!!)
+                    }
+                }
+                true
+            }
+            popup.show()
+        }
+    }
+
+    /**
+     *  Allows the user to delete an item from the realm.
+     */
+    private fun deleteItem(id: ObjectId) {
+        val config = SyncConfiguration.Builder(user, partition).build()
```

Comment:
I wonder if the correct pattern would actually be to set the default realm configuration when you log in. Then you can get the default realm from here without needing to know the user. Another option would be to pass in the configuration rather than the user and partition. That would make this adapter more flexible.

Reply:
Yeah, setting the default realm in somewhere in this project makes a lot of sense, since generally you're only using one realm in this app. +1.